src/examples/mnist/mnist_ffnn.cpp

```diff
@@ -24,7 +24,7 @@ const std::vector<std::string> VALID_SET
 using namespace marian;
 
 int main(int argc, char** argv) {
-  auto options = New<Config>(argc, argv, ConfigMode::training, false);
+  auto options = New<Config>(argc, argv, cli::mode::training, false);
```

Comment:
Are we mixing abstraction levels? Should this not be Config::mode::training, rather than cli::? cli seems to be the external parser library that we use, while 'training' clearly refers to a Marian concept, and Config has knowledge of Marian.

Reply:
Was thinking about that, but the only place where this is actually used is CLI options. Once we start rewriting the whole options/config chaos you addressed earlier, we should be able to do away with this by having hierarchical options.

Reply:
We could import the cli::mode into Config::mode? E.g. in Config, using cli::mode or so?

Reply:
I added a TODO related to this resolving one of the previous suggestions.